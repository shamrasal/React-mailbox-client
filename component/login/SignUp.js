import { Link, useHistory } from "react-router-dom";
import React, { useRef } from "react";
import { Fragment } from "react";
import classes from "./SignUp.module.css";

const SignUp = () => {
  const history = useHistory();
  const enteredEmailRef = useRef();
  const enteredPasswordRef = useRef();
  const confirmPasswordRef = useRef();

  const submitHandler = (event) => {
    event.preventDefault();
    const enteredEmail = enteredEmailRef.current.value;
    const enteredPassword = enteredPasswordRef.current.value;
    const confirmPassword = confirmPasswordRef.current.value;

    if (enteredPassword === confirmPassword) {
      fetch(
        "https://identitytoolkit.googleapis.com/v1/accounts:signUp?key=AIzaSyAsRF3UsCeuJuSJzmWw5-xvKVUiR1QOWd0",
        {
          method: "POST",
          body: JSON.stringify({
            email: enteredEmail,
            password: enteredPassword,
            returnSecureToken: true,
          }),
        }
      )
        .then((res) => {
          if (res.ok) {
            res.json().then((data) => {
              console.log(data);
              console.log("User has successfully signed up.");
              history.replace("/signin");
            });
          } else {
            return res.json().then((data) => {
              console.log(data);
              let errorMessage = "authentication failed...";
              if (data && data.error && data.error.message) {
                errorMessage = data.error.message;
              }
              console.log(errorMessage);
              alert(errorMessage);
            });
          }
        })
        .then((data) => {
          console.log(data);
        })
        .catch((err) => {
          console.log(err);
        });
      enteredEmailRef.current.value = "";
      enteredPasswordRef.current.value = "";
      confirmPasswordRef.current.value = "";
    } else {
      alert("password not match...");
      enteredEmailRef.current.value = "";
      enteredPasswordRef.current.value = "";
      confirmPasswordRef.current.value = "";
    }
  };
  return (
    <Fragment>
      <section className={classes.auth}>
        <h1>{"Sign Up"}</h1>
        <form onSubmit={submitHandler}>
          <div className={classes.control}>
            <label htmlFor='email'>Email</label>
            <input type='email' id='email' ref={enteredEmailRef} required />
          </div>
          <div className={classes.control}>
            <label htmlFor='password'>Password</label>
            <input
              type='password'
              id='password'
              ref={enteredPasswordRef}
              required
            />
          </div>
          <div className={classes.control}>
            <label htmlFor='password'>Confirm Password</label>
            <input
              type='password'
              id='password1'
              ref={confirmPasswordRef}
              required
            />
          </div>
          <div className={classes.actions}>
            <button>"Sign Up"</button>
            <Link to={"/signin"}>Already have a account? Sign In</Link>
          </div>
        </form>
      </section>
    </Fragment>
  );
};
export default SignUp;
