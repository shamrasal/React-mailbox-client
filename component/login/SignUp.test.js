import { render, screen } from "@testing-library/react";
import SignUp from "./SIgnUp";

test("render sign up test", () => {
  render(<SignUp />);
  const buttonElement = screen.getByRole("button");
  expect(buttonElement).toMatchSnapshot();
});
