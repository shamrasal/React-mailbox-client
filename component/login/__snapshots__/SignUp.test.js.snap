// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render sign up test 1`] = `
<button>
  "Sign UP"
</button>
`;
